### ISSUE TITLE
our embedded xml tags in the plain-text edition

### ISSUE THREAD

**Issue by ebeshero** [ebeshero/Pittsburgh_Frankenstein#8]
@Rikkm I'm doing a round of work on 1831 that caused me to check how we'd prepped the same chunk of text in the 1818 and 1823. And I suddenly discovered something we marked differently in the 1818 than in the other two texts. I bet we talked about it, because we're doing things right in the 1823 and 1831, but we'll need to correct the 1818 for the following issue:

Here's a sample of the 1818, with hard returns around the `<pb/>` element:

````
text text text he embarks 

<pb xml:id="1818_v1_016" n="v1_004"/>

in a little boat, text text text 

````
This is problematic because hard returns actually signal new paragraphs in our pseudo-plain text edition (that's the only signal we have in the plain text file for a new paragraph). So, what we want to do, as we're doing in the other two texts should be like this:

````
text text text he embarks<pb xml:id="1818_v1_016" n="v1_004"/> in a little boat, text text text 
````

I would be tempted to do a find and replace and remove all the returns around `<pb>` elements, **except** that some of them maybe (probably, almost certainly) **do** come at paragraph endings. I know that I can find all the `<pb/>`'s that come between words that start with lower-case letters, or that come after or before commas, semicolons, colons (etc), and that will catch most of these in an auto-correction, but where a page-break comes after an end-stopped punctuation mark, we need to check it against the 1818 photofacsimile. 

I wonder if I should stop and try to partially correct these right now, or save it for a pass of final proofing next week?  What think you?

**Comment by ebeshero** [comment]
Also, one thing I'm now explicitly altering across all three texts is the way we deal with footnotes to the poetry embedded in the novel, on the same principle of not disrupting the signaling of semantic paragraphs. I'm actually introducing a `<note>` element and positioning the notes right after what they annotate, rather than attempting to position them according to their page layout. This is a convention of the TEI for semantic encoding rather than prioritizing the layout of the page. (We can signal in an attribute that it is the author's footnote to the passage in question, and render it where we choose on the digital edition.)  

Basically, we need to make sure we're writing markup in a consistent way across the three texts so as to accomplish a clean and unambiguous collated document...which I'm eager to see...:-)

**Comment by ebeshero** [comment]
And...yikes! I've had to enter three whole lines of text that was missing in the PAEE 1831 edition! Good thing we're correcting these...

**Comment by Rikkm** [comment]
ok, now we have a problem.
I was editing 1823 and lost internet before I could post the most recent
changes.
We are talking a couple of hours of work.

Can I commit my 1823? Would you be able to redo the thing you were doing
across all three texts?

**Comment by ebeshero** [comment]
@Rikkm  Yes, I think you should commit your edits, and then pull in the changes. Then Git will do the thing it was made to do--and merge the versions together. You might get a merge conflict, but that's not a big deal to fix.

**Comment by ebeshero** [comment]
@Rikkm If you do get a merge conflict, post me on it here--I'll keep an eye out. (I've been walking my students through merge conflicts on their semester projects all week!)

**Comment by ebeshero** [comment]
@Rikkm More specifically: Here's the merge flow that I think should work:
1) commit your changes.
2) do a `git pull` (and note if that provokes a conflict, as in can't pull because the changes would overwrite...) If so, enter `git status` and follow the advice you get there...
3) Then presuming the pull eventually goes okay, do `git status` again, and follow its advice--it should tell you to do a `git push` to push your merged files complete with your updates to the remote. 

Hope that works!

**Comment by Rikkm** [comment]
i tried the commit - it aborted.


vanaheim:Pittsburgh_Frankenstein rikk$ git pull

Already up-to-date.

vanaheim:Pittsburgh_Frankenstein rikk$ git status

On branch Text_Processing

Your branch is up-to-date with 'origin/Text_Processing'.

Changes not staged for commit:

  (use "git add <file>..." to update what will be committed)

  (use "git checkout -- <file>..." to discard changes in working directory)

modified:   Plain_Texts/1823_full.txt

no changes added to commit (use "git add" and/or "git commit -a")

vanaheim:Pittsburgh_Frankenstein rikk$ git pull

remote: Counting objects: 19, done.

remote: Compressing objects: 100% (19/19), done.

remote: Total 19 (delta 9), reused 0 (delta 0), pack-reused 0

Unpacking objects: 100% (19/19), done.

From https://github.com/ebeshero/Pittsburgh_Frankenstein

   49c5890..46fba21  Text_Processing -> origin/Text_Processing

   6f0df85..46fba21  ebeshero_Exp    -> origin/ebeshero_Exp

   20648d3..34cf052  master          -> origin/master

Updating 49c5890..46fba21

*error: Your local changes to the following files would be overwritten by
merge:*

* Plain_Texts/1823_full.txt*

Please commit your changes or stash them before you merge.

Aborting

vanaheim:Pittsburgh_Frankenstein rikk$ git pull

Updating 49c5890..46fba21

error: Your local changes to the following files would be overwritten by
merge:

Plain_Texts/1823_full.txt

Please commit your changes or stash them before you merge.

Aborting

vanaheim:Pittsburgh_Frankenstein rikk$ git commit -m "edits to 1823 let us
see what we need to fix?"

On branch Text_Processing

Your branch is behind 'origin/Text_Processing' by 3 commits, and can be
fast-forwarded.

  (use "git pull" to update your local branch)

Changes not staged for commit:

modified:   Plain_Texts/1823_full.txt


no changes added to commit

vanaheim:Pittsburgh_Frankenstein rikk$ git pull

Updating 49c5890..46fba21

error: Your local changes to the following files would be overwritten by
merge:

Plain_Texts/1823_full.txt

Please commit your changes or stash them before you merge.

Aborting

vanaheim:Pittsburgh_Frankenstein rikk$


so how do I "stash" them?

**Comment by ebeshero** [comment]
You don't want to stash your changes (that means setting them aside). We *want* your changes. 
It looks like your commit didn't go through. Try this:

`git status` 
and report what you read.

**Comment by ebeshero** [comment]
I think the next step is actually going to be:
`git commit -a -m "your commit message here"` 
to make sure your commit is entered...

**Comment by Rikkm** [comment]
On branch Text_Processing

Your branch is behind 'origin/Text_Processing' by 3 commits, and can be
fast-forwarded.

  (use "git pull" to update your local branch)

Changes not staged for commit:

  (use "git add <file>..." to update what will be committed)

  (use "git checkout -- <file>..." to discard changes in working directory)

modified:   Plain_Texts/1823_full.txt

no changes added to commit (use "git add" and/or "git commit -a")

vanaheim:Pittsburgh_Frankenstein rikk$

**Comment by djbpitt** [comment]
"git stash" puts them on a shelf while you pull. Then you can retrieve and
apply them

On Apr 27, 2017, at 9:02 PM, Elisa Beshero-Bondar <notifications@github.com>
wrote:

You don't want to stash your changes (that means setting them aside). We
*want* your changes.
It looks like your commit didn't go through. Try this:

git status
and report what you read.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/ebeshero/Pittsburgh_Frankenstein/issues/8#issuecomment-297879465>,
or mute the thread

**Comment by ebeshero** [comment]
@djbpitt I guess I've done this differently. I've been committing first, then doing the merge.

**Comment by ebeshero** [comment]
@Rikkm what happens when you:
`git commit -a -m "enter a commit message"`

You should be able to accomplish the commit.

**Comment by djbpitt** [comment]
I think that has the same effect.

On Apr 27, 2017, at 9:06 PM, Elisa Beshero-Bondar <notifications@github.com>
wrote:

@djbpitt <https://github.com/djbpitt> I guess I've done this differently.
I've been committing first, then doing the merge.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ebeshero/Pittsburgh_Frankenstein/issues/8#issuecomment-297880056>,
or mute the thread

**Comment by Rikkm** [comment]
i've been following the process you outlined:

git status

do this when you don’t know what to do next!

git add .

adds git tracking to the new stuff

git commit -a -m “your specific pithy commit message here"

git commit -m “message”

git push


this is what I get after git status now:

On branch Text_Processing

Your branch and 'origin/Text_Processing' have diverged,

and have 1 and 3 different commits each, respectively.

  (use "git pull" to merge the remote branch into yours)

All conflicts fixed but you are still merging.

  (use "git commit" to conclude merge)

Changes to be committed:

new file:   Facsimile_PrintPubs/1831_missingPgs.pdf

modified:   Plain_Texts/1818_full.txt

modified:   Plain_Texts/1823_full.txt

modified:   Plain_Texts/1831_full.txt

**Comment by ebeshero** [comment]
@Rikkm Yay! That's hopeful. Time for a
`git pull`

**Comment by Rikkm** [comment]
working on it

**Comment by ebeshero** [comment]
Errr...wait a second...I mean, follow git's instructions

**Comment by ebeshero** [comment]
which are one more commit!!!

**Comment by ebeshero** [comment]
For this commit message, I'd say, `git commit -a -m "resolving merge conflict"`

**Comment by ebeshero** [comment]
(If you hit `git pull` on my mark, don't worry--it won't pull if it can't deal with the merge...)

**Comment by Rikkm** [comment]
trying but it is....hanging

**Comment by ebeshero** [comment]
@Rikkm If it's hanging, close the terminal, and reopen. And do `git status`.

**Comment by Rikkm** [comment]
latest:

vanaheim:Pittsburgh_Frankenstein rikk$ git pull

error: You have not concluded your merge (MERGE_HEAD exists).

hint: Please, commit your changes before merging.

fatal: Exiting because of unfinished merge.

vanaheim:Pittsburgh_Frankenstein rikk$ git commit -m "RM 1823 again again"

[Text_Processing bd61af4] RM 1823 again again

vanaheim:Pittsburgh_Frankenstein rikk$

**Comment by Rikkm** [comment]
i think, finally, i have it

**Comment by ebeshero** [comment]
@Rikkm No worries. Git status solves all. ;-)

**Comment by Rikkm** [comment]
nothing to commit, working tree clean

**Comment by ebeshero** [comment]
Hmmm... time for that `git pull` at last?

**Comment by Rikkm** [comment]
new note:

On branch Text_Processing

Your branch is ahead of 'origin/Text_Processing' by 2 commits.

  (use "git push" to publish your local commits)

nothing to commit, working tree clean


 -- oops, so now I push

**Comment by ebeshero** [comment]
YES. `git push` please. Let's get your changes into the remote branch.

**Comment by Rikkm** [comment]
push done. status up to date.
- finally

**Comment by mjlavin80** [comment]
https://cdn.meme.am/instances/66079643.jpg

**Comment by ebeshero** [comment]
Now, I wonder what happens if you do a `git pull`? (If you already managed to pull in everything, you're through and there won't be anything to pull, so it can't do any harm.)

**Comment by ebeshero** [comment]
@mjlavin80 lolz!!!!

**Comment by ebeshero** [comment]
@Rikkm Meanwhile, I'm gonna pull in your changes and take a look at what happened after the dust settles... ;-) Ahhh, I love a good merge conflict.

**Comment by Rikkm** [comment]
pull done again - already up to date response

**Comment by ebeshero** [comment]
Huzzah! All is well. Everybody who wants an up-to-date repo, go pull, although @mjlavin80 and @djbpitt I think none of this affected the master branch--just the land of Text_Processing where we're frantically coding away...

**Comment by Rikkm** [comment]
I'm up against three deadlines, so I'm juggling projects.

Now I need to hunt down the issues you raised and copy them into one doc.

With that in mind, what do I still have to fix in 1818 or need to change
about how I am doing 1823.
I think it is the hard returns, but was there something else. I now have a
LOT of emails.

**Comment by ebeshero** [comment]
@Rikkm Go to this website: https://github.com/ebeshero/Pittsburgh_Frankenstein/issues/8#issuecomment-297877112 where all is posted in a threaded Issue post. My observations about the diff btw 1818 and the other texts are just posted at the top of this issue. I can make it a single wiki in markdown.

You do *not* need to change anything you're doing in 1823. It's just an old way we must have been doing things back in January on the 1818 text, and I think I can fix most of it, but we're going to need to proof it carefully against the facsimile for paragraph breaks around our inserted `<pb/> elements

**Comment by ebeshero** [comment]
How about I tidy up this issue, and just delete all the merge conflict stuff--*except* @mjlavin80's awesome link? ;-)

**Comment by Rikkm** [comment]
I can work with that.

what does this mean practically?
That I stay hands off until you ping me, then pull again?

**Comment by ebeshero** [comment]
No, hands on, please! Keep pulling and committing and pushing as usual--no need to stop anything you're doing.

**Comment by ebeshero** [comment]
@Rikkm I've pulled in the changes on Text_Processing, but I'm confused by your first commit. It looks like you moved `<pb/>` elements outside of the words, when page-breaks fell inside a word.

As I've been marking these, all along, I've place the `<pb/>` elements exactly where they fall, even when they fall inside a word. They're a distinct feature of each edition. But we've silently been deleting the hyphenation that falls around the page-breaks. Since we are preparing a collated edition, we're prioritizing semantic breaks rather than page layout: the page-breaks are incidental, but we keep them eventually to pair up facsimile images with the text. I hope we can put those `<pb/>` elements back exactly where they fall, if we're going to be using them.

**Comment by Rikkm** [comment]
Elisa -- umm -- we made a decision back in January to move the <pb> to the
end of the word and attached punctuation.
This is what we have listed in the EditingFrankenstein.md

5) Formerly hyphenated words that cross page breaks will be joined on the
previous page before the <pb> is added to the text file. Any immediate
punctuation will be retained, most typically commas and periods.

So, yes, I've been doing this purposely, and would need to go back and
speed check the ends of pages to correct if I misunderstood our policy.

**Comment by ebeshero** [comment]
@Rikkm That's not how I read point 5: 
````
5) Formerly hyphenated words that cross page breaks will be joined on the previous page before the <pb> is added to the text file. Any immediate punctuation will be retained, most typically commas and periods.
````
As I understood this, we are joining the hyphenated words, but still positioning the `<pb/>` inside them, because the `<pb/>` as a milestone element isn't disrupting anything, and we aren't using it for layout. We are only changing the punctuation, not changing where we indicate a page-break falls. That has been my understanding since we implemented that policy. And it's consistent with my practice on other TEI projects...

I'm sorry this turned out to be an ambiguous policy! What can we do to repair it? I think we need to round a proofing of those `<pb/>` elements anyway, across the board.

**Comment by Rikkm** [comment]
so I'm just making sure:

all the edits I just corrected in 1823, again, like:

page 1: ...Peters
page 2: burg...

I've been doing:
Petersburg,<pb>

But you're saying these should be:
Peters<pb>burg

in which case I need to fix 1818 and the entire first volume of 1823.

**Comment by ebeshero** [comment]
@Rikkm Yes, that's it. Here's what happens: 

Peters<pb xml:id="1818_v1_025"/>burg
is literally the same word as
Petersburg <pb xml:id="1823_v1_018"/>

and we instruct collateX to read around those tags, but we still want to keep them where they fall, if preserving page-break information from the three editions is important to us. Since we've gone to the trouble of positioning the `<pb/> elements, and we've been planning to pair images to text, we can use that information ultimately when we're building the web edition--and even if it looks weird, it's still indicating exactly what the hyphen at the end of a word on the page does. That hyphen is what we'd call "pseudomarkup" (or even markup) made by a typesetter to indicate exactly what our angle-bracket markup is doing.

**Comment by Rikkm** [comment]
wonderful.

ok.
I'll get to fixing them. It won't be tonight.

**Comment by ebeshero** [comment]
I think we should just take note at this point (right now), that we have interpreted `<pb/>` positioning differently, and table that until we get through correcting the texts. I can take on fixing the 1818 pagebreaks if you can do the 1823. I'm sorry we didn't notice this discrepancy in our methods sooner! 

Meanwhile, just getting corrected texts is a lot more important right now.

**Comment by Rikkm** [comment]
fair enough.

I can fix the 1823 as I complete it.
I had hoped to finish it this weekend, BUT -- we have a major meeting on
May 8. I'm part of a group giving a report on materials that have not yet
been completed by others in our group. So I need to split time because our
deadline for consolidating the slidedeck is next Thursday.

Sorry that I misunderstood -- I was thinking the <pb> would be an actual
pagebreak in the browser...

**Comment by ebeshero** [comment]
@Rikkm I know--it's a reasonable idea, and you're thinking about the web design. I guess I'm trained and honed to think of TEI as informational and descriptive--though at our NEH Institute in July, Hugh Cayless will almost certainly be showing us how we can preserve all that information we're going to be accumulating on the web--without oversimplifying the TEI by *converting* it to simpler forms in standard HTML elements. (For a sneak preview, see his [CeTEIcean](https://github.com/TEIC/CETEIcean) project!)

**Comment by Rikkm** [comment]
and thus I have A LOT to learn at the workshop.
I'm learning a lot by doing this work with you -- if things go one way I'm
hoping we might take on digital editions through our hoped-for library
press imprint at CMU. In which case our workflows and editorial processes
with benefit from our (your and mine) miscommunications.

I *still* need to get back to Python by mid-May or I may be screwed in July.

**Comment by ebeshero** [comment]
@Rikkm Agreed--and what's good about all this is even when we have an occasional slightly stressful mixup, we get it sorted out! :-)  And we all have a lot to learn from that July institute...it's very exciting. :-) 

Nearer-term, I'm balancing this around Pitt's end-of-semester grading, which is due next Wed. I'm purposely doing a binge of Frankenstein work now to help get us "over the hump"--if we can actually finish text correction of all three by (say) Thurs. night next week, I think I can try getting the collateX machine cranking on our edition and have some collations to show at our meeting on Sat. 6 May. That is the goal...but even if we can't, I need to get our texts collated and packaged up by the last week of May when I'll be visiting with Raff and Wendell in Maryland...I hope we can hand off our collated edition to Raff so he can work on pointing it into the Shelley-Godwin archive.
